spec/unittest.dd

```diff
@@ -8,87 +8,113 @@ $(GRAMMAR
 $(GNAME UnitTest):
     $(D unittest) $(GLINK2 statement, BlockStatement)
 )
+    $(P Unit tests are a builtin framework of test cases
+    applied to a module to determine if it is working properly.
+    A D program can be run with unit tests enabled or disabled.
+    )
 
-        $(P Unit tests are a series of test cases applied to a module to determine
-        if it is working properly. Ideally, unit tests should be run every
-        time a program is compiled.
-        )
+    $(P Unit tests are a special function defined like:)
 
-        $(P Unit tests are a special function defined like:)
+    ---
+    unittest
+    {
+        ...test code...
+    }
+    ---
 
-------
-unittest
-{
-    ...test code...
-}
-------
+    $(P Individual tests are specified in the unit test using $(DDSUBLINK spec/expression, AssertExpression)s.
+    Unlike $(I AssertExpression)s used elsewhere, the assert is not assumed to hold, and upon assert
+    failure the program is still in a defined state.
+    )
 
     $(P There can be any number of unit test functions in a module,
     including within struct, union and class declarations.
     They are executed in lexical order.
-    The order in which the modules are called to run their unit tests is
-    implementation defined.
-    Stylistically, a unit test for a function should appear immediately
-    following it.
     )
 
-    $(P A compiler switch, such as $(DDSUBLINK dmd, switch-unittest, $(B -unittest))
-    for $(B dmd), will
-    cause the unittest test code to be compiled and incorporated into
-    the resulting executable. The unittest code gets run after
-    static initialization is run and before the $(D main())
-    function is called.
+    $(P Unit tests, when enabled, are run after all static initialization is
+    complete and before the $(D main()) function is called.
```

Comment:
This will change soon:
- https://dlang.org/changelog/2.078.0.html#core-runtime-unittestEnhancement
- https://dlang.org/spec/unittest.html

>  "test-or-main": any unit tests present will cause the program to summarize the results and exit regardless of the result. This will be the default in 2.080.

Reply:
When it changes, the spec should be updated.